packages/lodestar/src/util/sleep.ts

```diff
@@ -1,3 +1,30 @@
-export function sleep(ms: number): Promise<void> {
-  return new Promise((resolve) => setTimeout(resolve, ms));
+import {AbortSignal} from "abort-controller";
+import {ErrorAborted} from "./errors";
+
+/**
+ * Abortable sleep function. Cleans everything on all cases preventing leaks
+ * On abort throws ErrorAborted
+ */
+export async function sleep(ms: number, signal?: AbortSignal): Promise<void> {
```

Comment:
Maybe this function should go in lodestar-utils ? we can use it in the validator and other places